src/video/wayland/SDL_waylandwindow.c

```diff
@@ -102,6 +129,24 @@ SetFullscreen(SDL_Window *window, struct wl_output *output)
      */
     CommitMinMaxDimensions(window);
 
+#ifdef HAVE_LIBDECOR_H
+    if (viddata->shell.libdecor) {
+        if (wind->shell_surface.libdecor.frame == NULL) {
+            return; /* Can't do anything yet, wait for ShowWindow */
+        }
+        if (output) {
+            if (!(window->flags & SDL_WINDOW_RESIZABLE)) {
+                libdecor_frame_set_capabilities(wind->shell_surface.libdecor.frame, LIBDECOR_ACTION_RESIZE);
```

Comment:
IIUC you need to set `LIBDECOR_ACTION_RESIZE` before entry into full-screen (even if the window was not configured to be resizable by the user). When you un-fullscreen, you unset `LIBDECOR_ACTION_RESIZE` if the window was not meant to be resizable.

It may be worth a comment to that effect, because setting `LIBDECOR_ACTION_RESIZE` when `!(window->flags & SDL_WINDOW_RESIZABLE)` seems like a mistake on first reading.